src/module/Provider.sol

```diff
@@ -28,10 +28,19 @@ contract Provider is IProvider {
     error UnsupportedAsset(address asset);
     error RateIsNegative();
 
+    // precompute to save gas
+    bytes32 private constant _VERSION_0_1_1 = keccak256("0.1.1");
+    bytes32 private constant _VERSION_0_1_0 = keccak256("0.1.0");
+
     function isETHStrategyVault(address asset) public view returns (bool) {
         try IBaseStrategy(asset).STRATEGY_VERSION() returns (string memory version) {
             address vaultAsset = IVault(asset).asset();
-            return keccak256(bytes(version)) == keccak256(bytes("0.1.0")) && vaultAsset == MC.WETH;
+            if (vaultAsset != MC.WETH) {
+                // return early to save gas
+                return false;
+            }
+            bytes32 versionHash = keccak256(bytes(version));
+            return versionHash == _VERSION_0_1_0 || versionHash == _VERSION_0_1_1;
```

Comment:
let's require all strategies are upgraded

Reply:
fixed in [d798203](https://github.com/yieldnest/yieldnest-vault/pull/119/commits/d79820377c0ba701d4fcf51a4dc04d76652724ec)